src/client/js/highlight-category.js

```diff
@@ -15,13 +15,15 @@
  *  See the License for the specific language governing permissions and
  *  limitations under the License
  *
- */
+*/
 
 export default function highlightCurrentCategory() {
   const navLinks = document.querySelectorAll('nav a');
-  const category = document.querySelector('main').dataset.category;
-
-  for (const navLink of navLinks) {
-    navLink.classList.toggle('current', navLink.href.includes(category));
+  const main = document.querySelector('main');
+  if (main && main.dataset && main.dataset.category) {
```

Comment:
I'm not fond of silencing this error. 

The category highlight code relies on the presence of "main" currently. If there is no main, the code complains - and this is good, because it shows there is a problem of missing main and it should be fixed for the categories to work right...

Reply:
This is meant to cope with pages that don't have a category (like the 404 or cart page) or possibly page templates that don't want to use `<main>` (and/or provide a category).

Reply:
There should be no templates without main. Unless I'm not aware of some use case?

Reply:
Mandating `<main>` is sensible, but that doesn't fix the problem here, which is a lack of `data-category` attribute.

Do you think we should mandate that?

Reply:
If there is no data-category, main.dataset.category is undefined, and navLink.href.includes(category) resolves to false, and the code in master works just fine.

Reply:
You're right: as long as `<main>` is mandated, there's not a problem. `main.dataset.nonexistent` is `undefined`, i.e. not an error, as long as `main` exists.

Adjusted code back to what it was (i.e. without the test).